src/prefect/tasks.py

```diff
@@ -303,10 +304,14 @@ def __init__(
         if retry_jitter_factor is not None and retry_jitter_factor < 0:
             raise ValueError("`retry_jitter_factor` must be >= 0.")
 
-        self.retry_jitter_factor = retry_jitter_factor
+        self.result_storage = (
+            storage
+            if (storage := PREFECT_DEFAULT_RESULT_STORAGE_BLOCK.value()) is not None
+            else result_storage
+        )
```

Comment:
I think this is redundant with the change to `get_default_result_storage` and could be removed.